devops/tests/test_component.py

```diff
@@ -125,9 +125,28 @@ def test_patch_cronjob():
 def test_handling_long_nested_image_name():
     deploy = get_deployment()
     c = Component("service/test-service")
-    c.image = "gcr.io/google-containers/etcd-amd64:3.3.10-1"
+    c.image = "gcr.io/google-containers/etcd-amd64"
     c.image_pull_secrets = {"gcr.io": "secret"}
 
     c._patch_image_pull_secrets(deploy)
     spec = deploy["spec"]["template"]["spec"]
     assert spec["imagePullSecrets"][0]["name"] == "secret"
+
+
+def test_patch_containers_with_default_tag():
+    deploy = get_deployment()
+    c = Component("service/test-service")
+    c.image_prefix = ""
+
+    c._patch_containers(deploy)
+    container = deploy["spec"]["template"]["spec"]["containers"][0]
+    assert (
+        container["image"] == "imagined.registry.tld/myproj-service-test-deployment:v1"
+    )
+
```

Comment:
Yeah, I thought that `_get_full_docker_name` may cause some troubles due to inconsistency.. But my assumption was that `Component` instance doesn't know anything about values in the YAML, e.g `tag` is always set somewhere outside. That's why I guessed `_get_full_docker_name` won't be _logically_ affected by this change. 

IMO, to make `_get_full_docker_name` working with the tag from YAML, we should create a `Component` with the tag set to this value, instead of patching it somewhere in the middle of the process. To be honest I don't like side effects, they are difficult to debug

Reply:
I'm not really sure what is being said here.

Will `post_release.sh` not be executed when this is run in the expected manner, as in the image and tag are passed along from the CLI, or not?

Is there a special case where it does not work, like if you run release without specifying the image and tag?

If the issue is the first one - it must really be fixed.

If the issue is the second one - we can either
 - fix it by reading the YAML configs if available - though this would likely be challenging
 - make the parameters required
 - make the command give a warning when those parameters are missing - `Cannot reliably run post-release.sh, both x and y arguments are required`
 - just open an issue about it

Also, generally speaking it is expected throughout the project that the image etc. names can be largely parsed from the path, and that the settings have enough other information in them, about the hosts etc. and really only the version identifier should come from the build & release pipeline. Right?

The exception to this likely would be components hosted externally on Docker hub or whatever, but it's not likely they would have a `post-release.sh` script in them intended for use with this tool.